src/main/java/net/javadiscord/javabot/systems/starboard/StarboardManager.java

```diff
@@ -137,13 +141,18 @@ private void addMessageToStarboard(Message message, int stars, @NotNull Starboar
 				.sendMessage(String.format("%s %s", config.getEmojis().get(0), stars))
 				.setActionRow(Button.link(message.getJumpUrl(), "Jump to Message"))
 				.setEmbeds(embed);
-		for (Message.Attachment a : message.getAttachments()) {
-			try {
-				action.addFiles(FileUpload.fromData(a.getProxy().download().get(), a.getFileName()));
-			} catch (InterruptedException | ExecutionException e) {
-				action.addContent("Could not add Attachment: " + a.getFileName());
+		List<Attachment> attachments = message.getAttachments();
+		
+		if(attachments.size() > 1) {
+			for (Message.Attachment a : attachments) {
+				try {
+					action.addFiles(FileUpload.fromData(a.getProxy().download().get(), a.getFileName()));
+				} catch (InterruptedException | ExecutionException e) {
+					action.addContent("Could not add Attachment: " + a.getFileName());
```

Comment:
Re-interrupt the thread when an `InterruptedException` occurs.

Reply:
https://github.com/Java-Discord/JavaBot/pull/449#discussion_r1359447444

Reply:
Can you please still extract the `InterruptedException` catch block and re-interrupt the thread using `Thread.currentThread().interrupt()` as well use `ExceptionLogger.capture` for saving the exeption?